pkg/traceql/storage.go

```diff
@@ -175,6 +172,7 @@ func (s *Spanset) Release() {
 
 func (s *Spanset) clone() *Spanset {
 	ss := *s
+	ss.ServiceStats = maps.Clone(s.ServiceStats)
```

Comment:
not 100% sure if a deep clone is required here

Reply:
The SpanSet also contains the other data structures like `Attributes []*SpansetAttribute` and `Spans []Span`. Both are passed to `ss` without creating a deep clone first. I think it would be more consistent to do the same for `ServiceStats`.

I also checked all the parts where `s.clone()` is called and it does not look like a deep clone is required here

Reply:
Thanks for checking! I removed the maps.Clone().